src/java_bytecode/java_utils.cpp

```diff
@@ -182,3 +184,16 @@ dereference_exprt checked_dereference(const exprt &expr, const typet &type)
   result.set(ID_java_member_access, true);
   return result;
 }
+
+void java_add_components_to_class(
+  symbolt &class_symbol,
+  const struct_union_typet::componentst &components_to_add)
+{
+  struct_typet &struct_type=to_struct_type(class_symbol.type);
+  struct_typet::componentst &components=struct_type.components();
+
+  for(const struct_union_typet::componentt &component : components_to_add)
+  {
+    components.insert(components.begin(), component);
```

Comment:
Why not just `push_back`? Seems surprising to insert them back to front?

Reply:
This was actually me copying the implementation of `java_root_class`. I didn't understand why `insert` was used instead of `push_back`, but I assumed there must be a reason that the original author did that, so I thought I would roll with that. I can change it very easily, and I thought about it, but does it actually impact anything?

Reply:
I can't speak for `java_root_class` but following the principle of least surprise I think it makes sense to insert the components in order. I agree it shouldn't make any difference